Web.md

```diff
@@ -90,48 +74,133 @@ dictionary WebAssemblyInstantiatedSource {
+      1. Otherwise, if compilation fails, return a failure with the failure reason.
+   - Success steps, given a tuple (_module_, _instance_, _exports_)
+      1. Create _exportsObject_ from _exports_, in the same fashion as specified for the
+         [`WebAssembly.Instance`
+         constructor](https://github.com/WebAssembly/design/blob/master/JS.md#webassemblyinstance-constructor).
+      1. Create _instanceObject_ from _exportsObject_ and _instance_, in the same fashion as specified for the
+         [`WebAssembly.Instance`
+         constructor](https://github.com/WebAssembly/design/blob/master/JS.md#webassemblyinstance-constructor).
+      1. Let _moduleObject_ be a new `WebAssembly.Module` instance, with its [[Module]] set to _module_.
+      1. Let _result_ be [ObjectCreate](https://tc39.github.io/ecma262/#sec-objectcreate)([%ObjectPrototype%](https://tc39.github.io/ecma262/#sec-properties-of-the-object-prototype-object)).
+      1. Perform
+         [CreateDataProperty](https://tc39.github.io/ecma262/#sec-createdataproperty)(_result_,
+         "`module`", _moduleObject_).
+      1. Perform
+         [CreateDataProperty](https://tc39.github.io/ecma262/#sec-createdataproperty)(_result_,
+         "`instance`", _instanceObject_).
+      1. Return _result_.
+   - Failure steps, given _reason_
+      1. Return a new `WebAssembly.CompileError`, `WebAssembly.LinkError`, or
+         `WebAssembly.RuntimeError` as appropriate for the failure information stored in _reason_.
+
+#### Process a potential WebAssembly response
+
+The above two functions both reuse much of the same infrastructure for extracting bytes from an
+appropriate [`Response`](https://fetch.spec.whatwg.org/#response-class) object, differing only in
+what they do with those bytes in the end. As such we define the following shared spec-level
+procedure:
+
+Process a potential WebAssembly response accepts an input _argument_ and three sets of steps
+_processingSteps_, _successSteps_, and _failureSteps_:
+
+- _argument_ is an arbitrary author-supplied JavaScript value.
+- _processingSteps_ will occur [in
+  parallel](https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel) (i.e. "off the
+  main thread"), accepts a [byte sequence](https://infra.spec.whatwg.org/#byte-sequence), and must
+  return a Realm-agnostic success or failure value.
+- _successSteps_ and _failureSteps_ will occur in tasks posted back to the main event loop; they
+  accept the Realm-agnostic values returned from _processingSteps_, and must transform them into
+  JavaScript objects that can be used to appropriately fulfill or reject the returned promise.
+
+Given these values, to process a potential WebAssembly response:
+
+1. Let _returnValue_ be [a new
+   promise](https://www.w3.org/2001/tag/doc/promises-guide#a-new-promise).
+1. Let _sourceAsPromise_ be [a promise resolved
+   with](https://www.w3.org/2001/tag/doc/promises-guide#a-promise-resolved-with) _source_.
```

Comment:
given the input description above on new line 129, should "source" actually be "argument" in these variable names?